From: Marco Felsch <m.felsch@pengutronix.de>
Date: Fri, 22 Jul 2022 11:08:19 +0200
Subject: [PATCH] soc: imx: imx8mp-blk-ctrl: set the LCDIF hurry priority

Status: WIP, needs clarification with NXP first.

Set the LCDIF hurry priority to highest possible value so FIFO underruns
can be avoided. The hurry priority will be set by the BLKCTL hw as soon
as the LCDIF panic signal is active and set back to 'normal' priority
after the panic signal is released.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>
---
 drivers/soc/imx/imx8mp-blk-ctrl.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/soc/imx/imx8mp-blk-ctrl.c b/drivers/soc/imx/imx8mp-blk-ctrl.c
index 0f13853901df..e1694d403eef 100644
--- a/drivers/soc/imx/imx8mp-blk-ctrl.c
+++ b/drivers/soc/imx/imx8mp-blk-ctrl.c
@@ -201,6 +201,7 @@ static const struct imx8mp_blk_ctrl_data imx8mp_hsio_blk_ctl_dev_data = {
 #define HDMI_RTX_CLK_CTL3	0x70
 #define HDMI_RTX_CLK_CTL4	0x80
 #define HDMI_TX_CONTROL0	0x200
+#define  HDMI_LCDIF_NOC_HURRY_PRIO_MASK		GENMASK(14, 12)
 
 static void imx8mp_hdmi_blk_ctrl_power_on(struct imx8mp_blk_ctrl *bc,
 					  struct imx8mp_blk_ctrl_domain *domain)
@@ -217,6 +218,8 @@ static void imx8mp_hdmi_blk_ctrl_power_on(struct imx8mp_blk_ctrl *bc,
 		regmap_set_bits(bc->regmap, HDMI_RTX_CLK_CTL1, BIT(11));
 		regmap_set_bits(bc->regmap, HDMI_RTX_RESET_CTL0,
 				BIT(4) | BIT(5) | BIT(6));
+		regmap_set_bits(bc->regmap, HDMI_TX_CONTROL0,
+				FIELD_PREP(HDMI_LCDIF_NOC_HURRY_PRIO_MASK, 7));
 		break;
 	case IMX8MP_HDMIBLK_PD_PAI:
 		regmap_set_bits(bc->regmap, HDMI_RTX_CLK_CTL1, BIT(17));
